trackma/ui/gtk/mainview.py

```diff
@@ -740,13 +741,24 @@ def _view_context_menu(self, event):
         menu.popup_at_pointer(event)
 
     def _build_episode_menu(self, show):
-        total = show['total'] or utils.estimate_aired_episodes(show) or 0
+        library_episodes = set(self._engine.library().get(show['id'], ()))
+        total = show['total'] or max(
+            show['my_progress'],
+            *library_episodes,
+            utils.estimate_aired_episodes(show)
+        )
+        next_ep = show['my_progress'] + 1
 
         menu_eps = Gtk.Menu()
         for i in range(1, total + 1):
             mb_playep = Gtk.CheckMenuItem(str(i))
-            if i <= show['my_progress']:
-                mb_playep.set_active(True)
+            if i == next_ep:
+                mb_playep.set_label(str(i) + " - Next")
+                mb_playep.set_margin_left(10)
+                menu_eps.set_focus_child(mb_playep)
+            mb_playep.set_inconsistent(i < next_ep)
```

Comment:
I agree.

Reply:
> I'd probably switch that around and have the "inconsistent" state for not yet watched episodes while the watched ones are checked normally.

In isolation, i would agree, but i don't. Seeing `7/12` anywhere already states pretty obviously that you watched 7 episodes. It's not very useful to have the check boxes saying the same thing.
I did highlight the "next" episode, so it's very easy to find the boundary on shows with low episode count.

The inconsistent state is somewhat useful when you need to scroll down on shows with more than 100 episodes, because it's harder to find the next episode.

Reply:
> In isolation, i would agree, but i don't.

Please elaborate, I'm not following?

Reply:
"Outside" the context menu, you can already see the current show progress in multiple places. If we ignore those (take the context menu in isolation), i would agree with your suggestion too.

Reply:
I'm not sure we're talking about the same thing here.

What I proposed was to swap the "watched" with the "downloaded but not watched" styles because to me having something "watched" is equivalent to it being "completed" or "completely checked" whereas "downloaded but not watched" is sort of an intermediate state between "watched" and "not watched or downloaded".

Reply:
@imsamuka, what @FichteFoll meant is "inconsistent" style(`mb_playep.props.inconsistent`) should be used to indicate "downloaded but not watched" episodes,
 "active" style(`mb_playep.props.active`) should be used to indicate "watched" episodes.
OTOH, your approach is a bit counterintuitive as (watched?) should be `True` when you finished watching that episode, `False` if you didn't watch it, otherwise it'd be `Inconsistent`(in this case, you downloaded the episode, but didn't watch yet).

Reply:
@imsamuka this is the correct way(active: done, inactive: not done, inconsistent: in between)
![Peek 2022-06-10 22-07](https://user-images.githubusercontent.com/24864366/173109266-c8d6e328-c627-4d12-a691-45be267bba41.gif)

Reply:
And sorry for the late reply, I was busy for uni admission prep. today was the first one, a few more left(the next one is probably at the end of this month, I'll be down in a few days again...)

Reply:
@FichteFoll  @ahmubashshir I get what you mean.

My biggest point here, is that we don't need `mb_playep.props.active` to indicate "watched" episodes. When you see the "Next" episode, you can already tell that every episode before, was already watched. I used `mb_playep.props.inconsistent` to show it as well, but seeing how counterintuitive this design choice was, i propose to remove the inconsistent state completely and do something else:

```suggestion
            elif i > next_ep:
                mb_playep.set_margin_left(10)
```

![Peek 2022-06-11 14-29](https://user-images.githubusercontent.com/62220318/173198543-38abc9c2-e126-4856-bc73-6e9d8068b5f4.gif)

Now `mb_playep.props.active` only means "episode downloaded" and it's still pretty easy to find the "Next" episode (on shows with too many episodes) because of the indentation/left margin. No more "inconsistent" state messing everyone up.

I really wanted to make the "Next" episode already focused when you open the menu, so that you wouldn't need to scroll down, but i couldn't find how.

Reply:
> When you see the "Next" episode, you can already tell that every episode before, was already watched.

That is true, but if that bothers you or someone we need to talk about usage patterns of this menu in the first place to understand what it can be used for and how to best achieve that. 

- When you want to play the "next" episode only, then you should be using the separate menu entry for that.
- When you want to play an arbitrary episode of a show with <24 episodes, the menu does its job and displaying which episodes you have downloaded exactly is useful as the functionality wouldn't be available for the others. As such, it would be fine to remove the intermediary state again and instead just mark the downloaded episodes.
- When you want to play an arbitrary episode of a show with >24 episodes is bothersome and an input prompt would be much easier. This is also how it works on the CLI, which I use primarily.
- When you want to play an episode in close proximity to the last one you watched/the next episode, then the changes you made to the menu text are sufficient to make it stand out, alas only for shows with <24 episodes as anything else requires scrolling to find that one episode. In that case, a dialog as proposed previously would also work.

The actionale steps I would extract from this are:

1. Remove the "inconsistent" styling since the menu is primarily concerned about showing which episodes *can* be played and have been downloaded and we have an indicator already indicating the next episode.
2. Add a dialog prompt for shows with more than X episodes because the menu effectively becomes unusable. Take One Piece with over 1000 episodes, for example.

Reply:
Perfectly said. Regarding the 4 usage patterns, the fourth one is primarily my use case. More often than anything, I see myself playing the previous, or ~3 episodes before the next one. Actually, I don't now if there's anyone hoping to watch, like, 2 episodes *after* the next one...

> ...alas only for shows with <24 episodes as anything else requires scrolling to find that one episode.

As you can see in the GIF i sent, scrolling down 50, is still quite fast. Over 100 episodes, it becomes pretty annoying. If the menu was already scrolled down to the Next episode whenever opened, even in long lists, it would also be quite fine.

> Add a dialog prompt for shows with more than X episodes because the menu effectively becomes unusable. Take One Piece with over 1000 episodes, for example.

Yes, and since you can't see which episodes you have downloaded, i propose something like [this multi-column combo box](https://stackoverflow.com/questions/39184776/combobox-with-two-columns) instead of a simple input prompt. I will experiment with this right away.

There's also this idea:

![Multi-Column Submenu](https://user-images.githubusercontent.com/62220318/173241115-28035b40-fdaf-452d-b85b-3e3f66cf3453.png)


If even possible in GTK, using my GIF as a reference, it probably can accommodate up to ~130 episodes (26 * 5 columns) on screen at the same time, at that point, even One Piece would only take ~8 screens (1000 / 130). Still quite annoying to scroll down, but way more manageable. With more columns, it can be even better.

I would still use a single list in animes with <27 episodes, though.